api/src/services/setup/view-indexer.js

```diff
@@ -60,23 +82,27 @@ const getViewsToIndex = async () => {
  * @return {Promise}
  */
 const indexView = async (dbName, ddocId, viewName) => {
-  do {
-    try {
-      return await request.get({
-        uri: `${environment.serverUrl}/${dbName}/${ddocId}/_view/${viewName}`,
-        json: true,
-        qs: { limit: 1 },
-      });
-    } catch (requestError) {
-      if (!continueIndexing) {
-        return;
-      }
+  return await waitForRequest({
+    uri: `${environment.serverUrl}/${dbName}/${ddocId}/_view/${viewName}`,
+    json: true,
+    qs: { limit: 1 },
+  });
+};
 
-      if (!requestError || !requestError.error || !SOCKET_TIMEOUT_ERROR_CODE.includes(requestError.error.code)) {
-        throw requestError;
-      }
-    }
-  } while (continueIndexing);
+/**
+ * Returns a promise that resolves when a nouveau index is indexed.
+ * Retries querying the index until no error is thrown
+ * @param {String} dbName
+ * @param {String} ddocId
+ * @param {String} indexName
+ * @return {Promise}
+ */
+const indexNouveauIndex = async (dbName, ddocId, indexName) => {
+  return await waitForRequest({
+    uri: `${environment.serverUrl}/${dbName}/_design/${ddocId}/_nouveau/${indexName}`,
```

Comment:
issue: the `_design` part of this URL is already included in the ddocid!

```suggestion
    uri: `${environment.serverUrl}/${dbName}/${ddocId}/_nouveau/${indexName}`,
```

Reply:
So this means I wasn't testing the indexing properly...the thing is that nouveau indexes are built any time a design document is created or updated, so the index will be built and "warmed" even if nothing queries the endpoint. Which is what I was testing for.

Actually the point of querying the endpoint in upgrade-steps is to make sure that upgrade **waits** until that is actually done before swapping the staged and live ddoc.

Fixed the url and re-tested properly.

Reply:
Yeah, I think the [ddoc versioning issue](https://github.com/medic/cht-core/issues/9691#issuecomment-2790367531) was making this super tricky to test.  I actually only caught this because I noticed the api unexpectedly crashed the first time I switched to your branch (with a `404` error).  But, that error only happened for the _first_ time I ran the api server. If I restated it after the crash it would run fine (since the ddoc version in Couch was matching).